src/core_plugins/kibana/public/management/sections/indices/edit_index_pattern/scripted_fields_table/scripted_fields_table.js

```diff
@@ -1,138 +1,202 @@
+    };
+  }
+
+  componentWillMount() {
+    this.fetchFields();
+  }
+
+  fetchFields = async () => {
+    const fields = await this.props.indexPattern.getScriptedFields();
+
+    const deprecatedLangsInUse = [];
+    const deprecatedLangs = getDeprecatedScriptingLanguages();
+    const supportedLangs = getSupportedScriptingLanguages();
+
+    for (const { lang } of fields) {
+      if (deprecatedLangs.includes(lang) || !supportedLangs.includes(lang)) {
+        deprecatedLangsInUse.push(lang);
       }
+    }
+
+    this.setState({
+      fields,
+      deprecatedLangsInUse,
+      ...this.computeTableState(this.state.criteria, this.props, fields)
+    });
+  }
+
+  onDataCriteriaChange = criteria => {
+    this.setState(this.computeTableState(criteria));
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.fieldFilter !== nextProps.fieldFilter) {
+      this.setState(this.computeTableState(this.state.criteria, nextProps));
+    }
+    if (this.props.scriptedFieldLanguageFilter !== nextProps.scriptedFieldLanguageFilter) {
+      this.setState(this.computeTableState(this.state.criteria, nextProps));
+    }
+  }
+
+  computeTableState(criteria, props = this.props, fields = this.state.fields) {
+    let items = fields;
+    if (props.fieldFilter) {
+      const fieldFilter = props.fieldFilter.toLowerCase();
+      items = items.filter(field => field.name.toLowerCase().includes(fieldFilter));
+    }
+    if (props.scriptedFieldLanguageFilter) {
+      items = items.filter(field => field.lang === props.scriptedFieldLanguageFilter);
+    }
+
+    return getTableOfRecordsState(items, criteria);
+  }
+
+  renderCallOuts() {
+    const { deprecatedLangsInUse } = this.state;
+
+    return (
+      <CallOuts
+        deprecatedLangsInUse={deprecatedLangsInUse}
+        painlessDocLink={documentationLinks.scriptedFields.painless}
+      />
+    );
+  }
+
+  startDeleteField = field => {
+    this.setState({ fieldToDelete: field, isDeleteConfirmationModalVisible: true });
+  }
+
+  hideDeleteConfirmationModal = () => {
+    this.setState({ fieldToDelete: undefined, isDeleteConfirmationModalVisible: false });
+  }
+
+  deleteField = () =>  {
+    const { indexPattern } = this.props;
+    const { fieldToDelete } = this.state;
+
+    indexPattern.removeScriptedField(fieldToDelete.name);
+    this.fetchFields();
+    this.hideDeleteConfirmationModal();
+  }
+
+  renderDeleteConfirmationModal() {
+    const { fieldToDelete } = this.state;
+
+    if (!fieldToDelete) {
+      return null;
+    }
+
+    return (
+      <EuiOverlayMask>
+        <EuiConfirmModal
+          title={`Delete scripted field '${fieldToDelete.name}'?`}
+          onCancel={this.hideDeleteConfirmationModal}
+          onConfirm={this.deleteField}
+          cancelButtonText="Cancel"
+          confirmButtonText="Delete"
+          defaultFocusedButton={EUI_MODAL_CONFIRM_BUTTON}
+        />
+      </EuiOverlayMask>
+    );
+  }
+
+  renderNoFieldsFound() {
```

Comment:
The empty table looks a little weird:

![image](https://user-images.githubusercontent.com/1238659/35994861-907fa330-0cc6-11e8-9b23-8a5ceadbe9dc.png)

At a later point the Design team will come up with a nice solution for empty table states which we can use here.  For now can we just not render anything, including the table, when there are no fields?

Reply:
My comments for this screen:

Tab names should have init caps:

Fields
Scripted fields
Source filters

I feel that this paragraph is overwritten

These scripted fields are computed on the fly from your data.  They can be used in visualizations and displayed in your documents. However, they cannot be searched.  You can manage them here and add new ones as you see fit, but be careful, scripts can be tricky!

My recommendation:

You can use scripted fields in visualizations and display them in your documents.  However, you cannot search scripted fields. 


Button names use sentence case:

Add Scripted Field -> Add scripted field 

No scripted fields found. -> No scripted fields found (no ending punctuation)

Reply:
Thanks!

> 1)The delete button in the upper right is in red and the Delete button in the confirmation modal is blue. The color should be consistent. Red is typically used for Delete.

@cjcenizal Is this supported in EUI now? I don't see a prop that looks like it will do it. Want me to add something?